aeron-samples/src/main/java/io/aeron/samples/AeronStat.java

```diff
@@ -195,6 +211,28 @@ private static void workLoop(final long delayMs, final Runnable outputPrinter)
         while (running.get());
     }
 
+    private static void telegrafOutput(final CncFileReader cncFileReader, final StringJoiner telegrafMetric)
+    {
+        final CountersReader counters = cncFileReader.countersReader();
+        final String replaceRegex = "( |,|\\.|-)";
+        final Pattern pattern = Pattern.compile(replaceRegex);
+
+        counters.forEach(
+            (counterId, typeId, keyBuffer, label) ->
+            {
+                if (counterId <= 24)
```

Comment:
Would be better to specify a `SystemCounterDescriptor.SYSTEM_COUNTER_TYPE_ID == typeId`

Reply:
Besides the use of magic numbers being bad. The counter filter should be used here to be consistent with the existing output. We need to keep things consistent and respect the principle of least surprise.

Reply:
The existing output isn't compatible for example:
```
77:    1,617,392,357,597 - client-heartbeat: 499
 78:    1,617,392,357,726 - client-heartbeat: 496
 79:               61,600 - sub-pos: 488 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1 @48544
 80:    1,617,392,357,463 - client-heartbeat: 514
 81:               61,216 - pub-pos (sampled): 484 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1
 82:               61,600 - rcv-pos: 486 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 83:            8,450,208 - pub-lmt: 480 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 84:               61,600 - sub-pos: 497 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1 @48832
 85:               61,600 - sub-pos: 483 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1 @0
 86:               61,600 - snd-pos: 480 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 87:    1,617,392,357,963 - client-heartbeat: 487
 88:                    1 - rcv-channel: aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1 0.0.0.0:4336
 89:               61,600 - pub-pos (sampled): 480 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 90:               61,216 - rcv-hwm: 485 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1
 91:               61,216 - rcv-pos: 485 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1
 92:               61,216 - sub-pos: 481 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1 @0
 93:            8,449,824 - pub-lmt: 484 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1
 94:            2,158,752 - snd-lmt: 480 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 95:                    0 - snd-bpe: 480 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 96:               61,600 - rcv-hwm: 486 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1
 97:                    0 - snd-bpe: 484 -15765714 30 aeron:udp?endpoint=239.1.0.3:4336|interface=127.0.0.1
 98:    1,617,392,357,503 - client-heartbeat: 482
 99:    1,617,392,357,563 - client-heartbeat: 493
100:                    1 - snd-channel: aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1 0.0.0.0:4335
101:               61,600 - sub-pos: 512 -15765715 20 aeron:udp?endpoint=239.1.0.1:4335|interface=127.0.0.1 @51520
102:    1,617,392,357,701 - client-heartbeat: 511
```

The outputs here are sanitized by the regex filter above, but the other outputs contain keys that aren't valid for the input to telegraf:
```
======================================================================
  0:          118,496,480 - Bytes sent
  1:          118,448,256 - Bytes received
  2:                    0 - Failed offers to ReceiverProxy
  3:                    0 - Failed offers to SenderProxy
  4:                    0 - Failed offers to DriverConductorProxy
  5:                    0 - NAKs sent
  6:                    0 - NAKs received
  7:            1,843,606 - Status Messages sent
  8:            1,843,606 - Status Messages received
  9:            3,677,099 - Heartbeats sent
 10:            3,675,590 - Heartbeats received
 11:                    0 - Retransmits sent
 12:                    0 - Flow control under runs
 13:                    0 - Flow control over runs
 14:                    0 - Invalid packets
 15:                    0 - Errors
 16:                    0 - Short sends
 17:                    0 - Failed attempts to free log buffers
 18:                    0 - Sender flow control limits, i.e. back-pressure events
 19:                    0 - Unblocked Publications
 20:                    0 - Unblocked Control Commands
 21:                    0 - Possible TTL Asymmetry
 22:                    0 - ControllableIdleStrategy status
 23:                    0 - Loss gap fills
 24:                    2 - Client liveness timeouts
```

Reply:
If it cannot work with existing filters then it is not a good candidate to merge as it makes behaviour inconsistent.

Reply:
Any other suggestions on how to collect metrics? Parsing the existing metrics output with a wrapper is also a rough solution.

Reply:
It would make more sense to create a new tool similar to AeronStat the produces json output for metrics collection. AeronStat is a sample of how to build tools.